asdfghjkl/operations/operation.py

```diff
@@ -26,6 +26,7 @@ def __init__(self, module, model, op_names, save_field='op_results'):
         self._save_field = save_field
         self._grads_scale = None
         self._kron_A = None
+        self._unit_wise_A = None
```

Comment:
This field is used to store extended input data. 
This data is calculated in `forward_post_process` and is used in `backward_pre_process`.

One way to avoid storing the data in an extra field is by extracting Line53-63 in operation.py to a function. And we call this new function on demand.

Reply:
I see.
but both `_call_operations_in_forward()` and `_call_operatoins_in_backward()` receive the preprocessed `in_data`.
https://github.com/kazukiosawa/asdfghjkl/blob/126026ecdd01accfee9bcf328fce5f8cc2d7b420/asdfghjkl/core.py#L25-L31

so do we ready need this `_unit_wise-A`?

Reply:
Yes, the `in_data` of forward and backward are identical. But I want to get extended input. Now I create a new function called `extend_input` to do that and I will call it before `cov_unit_wise`.

Reply:
now I get your point! yes, preparing `extend_input()` sounds a reasonable solution